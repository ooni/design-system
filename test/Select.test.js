/* global describe, test, expect */
import React from 'react'
import { renderWithTheme } from './index'
import { Select } from '../'

describe('Select', () => {
  test('renders', () => {
    const { container } = renderWithTheme(
      <Select />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders with children', () => {
    const { container } = renderWithTheme(
      <Select>
        <option value="1">One</option>
        <option value="2">Two</option>
      </Select>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
