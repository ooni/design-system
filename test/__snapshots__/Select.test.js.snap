// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  background-color: #e9ecef;
  border: none;
  border-radius: 2em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  cursor: pointer;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-right: 40px;
  padding-left: 16px;
  width: unset;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
  width: 24px;
  height: 24px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0 c1"
>
  <select
    class="c2"
  />
  <svg
    class="c3"
    fill="currentcolor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;

exports[`Select renders with children 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: default;
  color: inherit;
  background-color: transparent;
  background-color: #e9ecef;
  border: none;
  border-radius: 2em;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  cursor: pointer;
  padding-top: 10px;
  padding-bottom: 10px;
  padding-right: 40px;
  padding-left: 16px;
  width: unset;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -28px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  pointer-events: none;
  width: 24px;
  height: 24px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="c0 c1"
>
  <select
    class="c2"
  >
    <option
      value="1"
    >
      One
    </option>
    <option
      value="2"
    >
      Two
    </option>
  </select>
  <svg
    class="c3"
    fill="currentcolor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;
